Sources/Basic/DictionaryExtensions.swift

```diff
@@ -0,0 +1,20 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension Dictionary {
```

Comment:
... which reminds me: I suspect there are a number of places we could use this in places where we have the following pattern:
```
var foo = Dictionary<>()
... initialize foo ...
self.foo = foo
```

Using this init instead makes it obvious that everything is immutable post-initialization.